### PR TITLE
Renovation: React compat radio group

### PR DESCRIPTION
**Pull request by const314**

### CODE REVIEW

File: packages/react/src/compatible-components/radio-group/radio-group.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+//------------------------------
+
+const createItemPropGetter = (propName: string) => (item: ItemLike) => (
+  Object.prototype.hasOwnProperty.call(item, propName) ? item[propName] : item
+);
+
+interface CompatibleRadioGroupProps<T> extends RadioGroupProps<T> {
+  items: Array<ItemLike>;
+  // eslint-disable-next-line react/require-default-props
+  itemRender?: (data: ItemLike, index?: number) => JSX.Element;
```
Comment (MikeVitik) [inline packages/react/src/compatible-components/radio-group/radio-group.tsx:21]: Add support for itemComponent

File: packages/react/src/compatible-components/radio-group/radio-group.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+}: CompatibleRadioGroupProps<RadioGroupValue>) {
+  const getItemLabel = createItemPropGetter(displayExpr || valuePropNameDefault);
+  const getItemValue = createItemPropGetter(valueExpr || valuePropNameDefault);
+
+  return (
+    <RadioGroup defaultValue={defaultValue}>
+      {items.map((item, index) => (
+        <RadioButton
+          value={getItemValue(item)}
+          label={itemRender ? itemRender(item, index) : getItemLabel(item)}
```
Comment (MikeVitik) [inline packages/react/src/compatible-components/radio-group/radio-group.tsx]: Move this logic into the radio button component and leave following:
```
<RadioButton
...
label={getItemLabel(item)}
labelTemplate={itemRender}
labelComponnet={itemComponent}
/>
```

File: packages/react/src/compatible-components/radio-group/radio-group.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  valueExpr,
+  displayExpr,
+}: CompatibleRadioGroupProps<RadioGroupValue>) {
+  const getItemLabel = createItemPropGetter(displayExpr || valuePropNameDefault);
+  const getItemValue = createItemPropGetter(valueExpr || valuePropNameDefault);
+
+  return (
+    <RadioGroup defaultValue={defaultValue}>
+      {items.map((item, index) => (
+        <RadioButton
```
Comment (MikeVitik) [inline packages/react/src/compatible-components/radio-group/radio-group.tsx]: You forget to add key

**Review by MikeVitik — CHANGES_REQUESTED** [review]

File: packages/react/src/compatible-components/radio-group/radio-group.tsx (hunk 2 above)
Comment (const314) [reply packages/react/src/compatible-components/radio-group/radio-group.tsx]: The types of renovated RadioButton and DevExtreme RadioGroup props are incompatible.
Old RadioGroup's labelComponent accepts { data } as props, where data is the item itself (of ItemLike type). 
RadioButton's labelTemplate accepts the label contents passed to RadioButton's label prop.
I think it is better to encapsulate all compatibility logic in the RadioGroupCompatible component.

File: packages/react/src/compatible-components/radio-group/radio-group.tsx (hunk 3 above)
Comment (const314) [reply packages/react/src/compatible-components/radio-group/radio-group.tsx]: fixed

File: packages/react/src/compatible-components/radio-group/radio-group.tsx (hunk 1 above)
Comment (const314) [reply packages/react/src/compatible-components/radio-group/radio-group.tsx:21]: fixed

File: playgrounds/react/src/examples/radio-group-compatible/radio-group-compatible-example.tsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
+import { AnyRecord } from '@devextreme/core';
+import { RadioGroupCompatible, RadioGroupValue } from '@devextreme/react';
+import { useState } from 'react';
+
+const nonStringOptions = [
```
Comment (IlyaKhD) [inline playgrounds/react/src/examples/radio-group-compatible/radio-group-compatible-example.tsx]: **Minor**:
```suggestion
const objectItems = [
```

File: packages/react/src/compatible-components/radio-group/radio-group.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+const valuePropNameDefault = 'text';
+
+export function RadioGroupCompatible({
+  items,
+  itemRender,
+  itemComponent: ItemComponent,
+  defaultValue,
+  valueExpr,
+  displayExpr,
+}: CompatibleRadioGroupProps<RadioGroupValue>) {
```
Comment (IlyaKhD) [inline packages/react/src/compatible-components/radio-group/radio-group.tsx]: Let's make it generic as early as now:
```suggestion
export function RadioGroupCompatible<TValue>({
  items,
  itemRender,
  itemComponent: ItemComponent,
  defaultValue,
  valueExpr,
  displayExpr,
}: CompatibleRadioGroupProps<TValue>) {
```

File: packages/react/src/compatible-components/radio-group/radio-group.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+  const renderLabel = (item: ItemLike, index: number) => {
+    if (ItemComponent) {
+      return (<ItemComponent data={item} />);
+    }
+    if (itemRender) {
+      return itemRender(item, index);
+    }
+    return getItemLabel(item);
+  };
```
Comment (IlyaKhD) [inline packages/react/src/compatible-components/radio-group/radio-group.tsx:51]: **Consider**:

```ts
type LabelRenderer = (item: ItemLike, index: number) => JSX.Element;
```

```suggestion
  const renderLabel: LabelRenderer =
    ItemComponent ? (item) => (<ItemComponent data={item} />)
      : itemRender ? (item, index) => itemRender(item, index)
        : (item) => getItemLabel(item)
```

Comment (const314) [reply packages/react/src/compatible-components/radio-group/radio-group.tsx:51]: Looks good, but we have https://eslint.org/docs/latest/rules/no-nested-ternary rule enabled.
Let's leave it as is for now.

File: playgrounds/react/src/examples/radio-group-compatible/radio-group-compatible-example.tsx (hunk 4 above)
Comment (const314) [reply playgrounds/react/src/examples/radio-group-compatible/radio-group-compatible-example.tsx]: fixed

File: packages/react/src/compatible-components/radio-group/radio-group.tsx (hunk 5 above)
Comment (const314) [reply packages/react/src/compatible-components/radio-group/radio-group.tsx]: fixed

File: packages/react/src/compatible-components/radio-group/radio-group.tsx (hunk 6 above)
Comment (IlyaKhD) [reply packages/react/src/compatible-components/radio-group/radio-group.tsx:51]: OK

**Review by IlyaKhD — APPROVED** [review]

**Review by williamvinogradov — APPROVED** [review]

**Review by MikeVitik — APPROVED** [review]

**Pull request merged**